Source/SwiftLintFramework/Rules/Style/VerticalWhitespaceClosingBracesRule.swift

```diff
@@ -0,0 +1,118 @@
+import Foundation
+import SourceKittenFramework
+
+private extension File {
+    func violatingRanges(for pattern: String) -> [NSRange] {
+        return match(pattern: pattern, excludingSyntaxKinds: SyntaxKind.commentAndStringKinds)
+    }
+}
+
+private extension Dictionary where Key: StringProtocol, Value: StringProtocol {
+    func cleanedKeysDict() -> [String: String] {
+        var cleanDict = [String: String]()
+
+        for (key, value) in self {
+            guard let keyString = key as? String else { continue }
+            let cleanedKey = keyString.replacingOccurrences(of: "↓", with: "")
+            cleanDict[cleanedKey] = value as? String
+        }
+
+        return cleanDict
+    }
+}
+
+public struct VerticalWhitespaceClosingBracesRule: ConfigurationProviderRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    private static let nonTriggeringExamples = [
+        "[1, 2].map { $0 }.filter {",
+        "[1, 2].map { $0 }.filter { num in",
+        """
+        /*
+            class X {
+
+                let x = 5
+
+            }
+        */
+        """
+    ]
+
+    private static let violatingToValidExamples: [String: String] = [
+        "    print(\"x is 5\")\n↓\n}": "    print(\"x is 5\")\n}",
+        "    print(\"x is 5\")\n↓\n\n}": "    print(\"x is 5\")\n}",
+        "    print(\"x is 5\")\n↓    \n}": "    print(\"x is 5\")\n}",
+        "        )\n}\n↓\n    }\n}": "        )\n}\n    }\n}",
+        "[\n1,\n2,\n3\n↓\n]": "[\n1,\n2,\n3\n]",
+        "foo(\nx: 5,\ny:6\n↓\n)": "foo(\nx: 5,\ny:6\n)",
+        "class Name {\n    run(5) { x in print(x) }\n↓\n}": "class Name {\n    run(5) { x in print(x) }\n}"
+    ]
+
+    private let pattern = "((?:\\n[ \\t]*)+)(\\n[ \\t]*[)}\\]])"
+}
+
+extension VerticalWhitespaceClosingBracesRule: OptInRule, AutomaticTestableRule {
+    public var configurationDescription: String { return "N/A" }
+
+    public init(configuration: Any) throws {}
+
+    public static let description = RuleDescription(
+        identifier: "vertical_whitespace_closing_braces",
+        name: "Vertical Whitespace before Closing Braces",
+        description: "Don't include vertical whitespace (empty line) before closing braces.",
+        kind: .style,
+        nonTriggeringExamples: (violatingToValidExamples.values + nonTriggeringExamples).sorted(),
+        triggeringExamples: Array(violatingToValidExamples.keys).sorted(),
+        corrections: violatingToValidExamples.cleanedKeysDict()
+    )
+
+    public func validate(file: File) -> [StyleViolation] {
+        let patternRegex: NSRegularExpression = regex(pattern)
+
+        return file.violatingRanges(for: pattern).map { violationRange in
+            let substring = file.contents.substring(from: violationRange.location, length: violationRange.length)
+            let substringRange = NSRange(location: 0, length: substring.count)
```

Comment:
I don't think this is unicode-safe. I think there was recently a helper added for this:

```suggestion
            let substringRange = substring.fullNSRange
```

Reply:
Done.